pusher-platform-core/src/main/kotlin/com/pusher/platform/subscription/BaseSubscription.kt

```diff
@@ -42,6 +44,10 @@ internal class BaseSubscription(
 
     private val job: ScheduledJob
 
+    private var activeResponseBody by Delegates.observable<ResponseBody?>(null) { _, old, _ ->
```

Comment:
I don't know if it can be made clearer what's going on here or if I'm just not familiar with the `Delegates` stuff but this `Delegates.observable<ResponseBody?>(null) { _, old, _ ->` looks mad to me 😀

Reply:
It ensures that if a new one is assigned, the old one is closed

Reply:
It's pretty magic (watching for reassignment of the variable). Can we not simplify it? Would testing for a current one and closing it before reassigning when we make a new one not work?

Reply:
Chances are this will not even be called more than once. It's more for defensive measures. I'll revert it to a simple nullable.